torch/library.py

```diff
@@ -567,6 +571,30 @@ def impl(qualname, types, func=None, *, lib=None):
         >>> x = torch.randn(3)
         >>> y = torch.ops.mylib.mysin(x)
         >>> assert torch.allclose(y, x.sin())
+        >>>
+        >>> # Example 2: Register function with decorator.
+        >>> def sin_decorator(func):
```

Comment:
Can you change the name of this? maybe "custom_decorator"? This doesn't have anything to do with torch.sin so it was confusing

Reply:
Implemented. Name changed to custom decorator and decorator refactored to func(*args, **kwargs) + 1. Also refactored the assert to account for the + 1.